src/SelfService/Domain/Models/MessageContractSchema.cs

```diff
@@ -29,6 +33,76 @@ public static MessageContractSchema Parse(string? text)
         throw new FormatException($"Value \"{text}\" is not valid.");
     }
 
+    public void CheckValidSchemaEnvelope()
+    {
+        CheckIsValidJsonSchema();
+
+        var jsonNode = JsonNode.Parse(_value)!.AsObject();
+        if (!jsonNode.TryGetPropertyValue("required", out var requiredPropertiesNode))
+            throw new FormatException($"Value \"{_value}\" is not valid, missing required key \"required\".");
+
+        var requiredKeys = requiredPropertiesNode?.AsArray().Select(x => x!.ToString()).ToList();
+        if (requiredKeys == null)
+            throw new FormatException($"Value \"{_value}\" is not valid, missing required key \"required\".");
+
+        string[] mandatoryEnvelopeKeys = { "schemaVersion", "type", "messageId" };
```

Comment:
This may be meta, but do we want to store our envelope definition in a schema too?

Reply:
i guess we could, but I dont know how to make such a schema